Source/SwiftLintFramework/Rules/ConditionalReturnsOnNewline.swift

```diff
@@ -23,22 +23,43 @@ public struct ConditionalReturnsOnNewline: ConfigurationProviderRule, Rule, OptI
             "guard true else {\n return true\n}",
             "guard true,\n let x = true else {\n return true\n}",
             "if true else {\n return true\n}",
-            "if true,\n let x = true else {\n return true\n}"
+            "if true,\n let x = true else {\n return true\n}",
+            "if textField.returnKeyType == .Next {",
+            "if true { // return }",
+            "/*if true { */ return }",
         ],
         triggeringExamples: [
             "guard true else { return }",
             "if true { return }",
             "if true { break } else { return }",
+            "if true { break } else {       return }",
         ]
     )
 
     public func validateFile(file: File) -> [StyleViolation] {
         let pattern = "(guard|if)[^\n]*return[^\n]\n*"
         let excludingKinds = SyntaxKind.commentAndStringKinds()
-        return file.matchPattern(pattern, excludingSyntaxKinds: excludingKinds).map {
+        return file.rangesAndTokensMatching(pattern).filter { range, tokens in
+            let kinds = tokens.flatMap { SyntaxKind(rawValue: $0.type) }
+
+            guard kinds.filter(excludingKinds.contains).isEmpty else {
+                return false
+            }
+
+            let nsString: NSString = file.contents
+            return !tokens.filter { token in
+                guard SyntaxKind(rawValue: token.type) == .Keyword else {
+                    return false
+                }
+
+                let tokenRange = NSRange(location: token.offset, length: token.length)
+                let tokenString = nsString.substringWithRange(tokenRange)
```

Comment:
`SyntaxToken`'s `offset` and `length` are byte based numbers.
We need to use SourceKittenFramework's byte based API's e.g. `NSString.substringWithByteRange(start:length:)`.